File: test_multilang/strip_test.go
// 剥离验证测试 — Go 版
// 注释和字符串中的控制流关键词不应被计数
//
// if thisIsComment { for i := 0; i < 10; i++ }
// fmt.Println("if you see this: for loop, while true, switch on")
// `raw string: if err != nil { for _, v := range items { switch v {} } }`
//
// Expected CC = 2 (base 1 + 1 real if)
package main

func StripTestTest(valid bool) string {
    // The only real control flow:
    if valid {  // CC +1 → real, total CC = 2
        return "ok"
    }
    // "if" "for" "while" "switch" ← all in strings, should NOT count
    return "no"
}

File: test_multilang/strip_test.js
/**
 * 剥离验证测试 — 注释和字符串中藏了大量 if/for/while/switch
 * 如果剥离失败 → CC 会高达 20+（误报）
 * 如果剥离成功 → CC ≈ 2（只计真实的 1 个 if）
 *
 * // if this is a comment, should NOT be counted
 * // for each mistake we make, don't count this either
 * console.log("if you see this string, the regex is blind"); // string
 * const msg = `while this template literal contains keywords`; // template literal
 * /* switch on this: block comment with if for while * /（去掉空格）
 */

function simpleCheck(valid) {
    // This is the ONLY real control flow
    if (valid) {           // CC +1 → total CC = 2
        return "ok";
    }
    // All comments above should NOT affect CC
    // "if you're counting these, your stripping is broken"
    return "not ok";
}
// Expected CC = 2 (base 1 + 1 real if)
